solr/solrj/src/java/org/apache/solr/client/solrj/cloud/autoscaling/AutoScalingConfig.java

```diff
@@ -305,10 +303,10 @@ public boolean equals(Object o) {
       return properties.equals(that.properties);
     }
 
-//    @Override
-//    public int hashCode() {
-//      throw new UnsupportedOperationException("TODO unimplemented");
-//    }
+    @Override
+    public int hashCode() {
+      return Objects.hash(name, actionClass);
```

Comment:
`equals()` is based on comparing `properties` and `hashCode()` on comparing `name` and `actionClass`. This can't guarantee two objets that are `equals()` are going to have the same `hashCode()`, as required.

Reply:
equals doesn't even include name, so I guess it'll just have to be properties only to conform to the contract.

Reply:
`name` comes from `properties` so would be ok to add it to the `hashCode()` computation, but it would not be useful if `properties` is already used.